### PR TITLE
sqlite_encryption - [iOS] Wallet encryption

### PR DESCRIPTION
**Pull request by TruszczynskiA**
## Description

- Implemented DB encryption functionality. Now, some fields in local sqlite3 DB will be encrypted. The encryption key (aka passphrase) is generated by the app.
- App will now remove temporary backup files when the app was unable to upload these files to iCloud.
- Added additional value stored in KeyChain - DBPassphraseKey
- Renamed BPKeychainWrapper to AppKeychainWrapper to better indicate the role of this element

## Motivation and Context
In our current implementation, local DB wasn't encrypted. To add an additional layer of safety, we decided to enable fields encryption in the local sqlite3 DB.

## How Has This Been Tested?
- This change was tested manually.

## Types of changes

* [ ] UI fix (non-breaking change which fixes a UI issue)
* [ ] Functionality bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [ ] I have tested this on multiple simulator devices with different screensizes.
* [x] I'm merging against the `development` branch.
* [x] Commits have been squashed into one commit with a descriptive commit message
* [x] I ran all tests before pushing.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added/changed tests to cover my changes if not UI changes.

### CODE REVIEW

File: MobileWallet/Common/Extensions/String.swift [hunk 1]
```diff
@@ -58,6 +58,16 @@ extension String {
             return nil
         }
     }
+
+    static func random(unicodeRange: Range<UInt8> = 0..<UInt8.max, lenght: Int) -> String {
+        (0..<lenght)
```
Comment (CjS77) [inline MobileWallet/Common/Extensions/String.swift:63]: typo

**Review by CjS77 — DISMISSED** [review]
LGTM - just one typo

**Review by CjS77 — DISMISSED** [review]
LGTM

**Review by mikethetike — APPROVED** [review]
Looks correct from my basic knowledge of swift. 

One question: So when the db is backed up to icloud, it is unencrypted via the wallet_ffi, and then encrypted as a backup? Seems like the correct choice to do it that way, as far as I see

**Comment by TruszczynskiA** [comment]
> Looks correct from my basic knowledge of swift.
> 
> One question: So when the db is backed up to icloud, it is unencrypted via the wallet_ffi, and then encrypted as a backup? Seems like the correct choice to do it that way, as far as I see

Yes. The FFI passphrase is generated and stored locally. That's why we need to decrypt it before we upload it to iCloud to prevent situation where user is unable to use the backup on a different device (with different passphrase).

**Pull request merged**